### ISSUE TITLE
Libtorch 1.12.1 crashes on debug mode under Windows

### ISSUE THREAD

**Issue by rodrigovimieiro** [pytorch/pytorch#83291]
### 🐛 Describe the bug

I am trying to run a simple Pytorch code using Libtorch 1.12.1. It runs perfectly under **Release** mode, but not with **Debug**. This problem happens when using the GPU. 

I had to add `/INCLUDE:"?ignore_this_library_placeholder@@YAHXZ"` on VS linker to make GPU works on 1.12.1. As stated [here](https://github.com/pytorch/pytorch/issues/54131).

When using an old Libtorch 1.8.2, the code runs fine on Release and Debug mode.

For 1.12.1 (Debug), It does load the model correctly but breaks in the  `module.forward` line.  

![01](https://user-images.githubusercontent.com/29632934/184226823-a957cd89-315e-4376-b212-c826d0113d36.png)

That is the sample code I am using:

```cpp
#include <torch/script.h> // One-stop header.

#include <iostream>
#include <string.h>
#include <memory>


using namespace std;

int main()
{

    string model_file = "UNet.pt";
    string img_file = "img.png";

    torch::Device device = torch::kCUDA;

    /*
     * ---- LOAD MODEL ----
    */

    torch::jit::script::Module module;
    try {
        // Deserialize the ScriptModule from a file using torch::jit::load().
        module = torch::jit::load(model_file, device);
    }
    catch (const c10::Error& e) {
        std::cerr << "error loading the model\n";
        std::cerr << e.msg() << std::endl;
        return -1;
    }
    std::cerr << "model loaded\n";

    
    /*
     * ---- Forward through model ----
    */

    at::Tensor tensor_image = torch::ones({ 1, 1, 224, 224});

    std::vector<torch::jit::IValue> tensor_image_jit;
    tensor_image_jit.push_back(tensor_image.to(device));
    
    // Execute the model and turn its output into a tensor.
    at::Tensor output = module.forward(tensor_image_jit).toTensor();
    output = output.to(torch::kCPU);
    output = output.toType(torch::kFloat32);

    std::cerr << "image processed\n";
    
    return 0;
}
```
Here is the Call stack:

 	ucrtbased.dll!issue_debug_notification(const wchar_t * const message) Line 28	C++
 	ucrtbased.dll!__acrt_report_runtime_error(const wchar_t * message) Line 154	C++
 	ucrtbased.dll!abort() Line 61	C++
 	ucrtbased.dll!terminate() Line 59	C++
 	vcruntime140_1d.dll!00007ff9aadf223a()	Unknown
 	vcruntime140_1d.dll!00007ff9aadf2ec5()	Unknown
 	vcruntime140_1d.dll!00007ff9aadf6ceb()	Unknown
 	ntdll.dll!00007ff9d16d229f()	Unknown
 	ntdll.dll!00007ff9d1681454()	Unknown
 	ntdll.dll!00007ff9d16d0dce()	Unknown
 	KernelBase.dll!00007ff9cf2f4fd9()	Unknown
 	vcruntime140d.dll!00007ff99219b460()	Unknown
 	c10.dll!c10::detail::torchCheckFail(const char * func, const char * file, unsigned int line, const std::string & msg) Line 87	C++
 	c10.dll!c10::detail::torchInternalAssertFail(const char * func, const char * file, unsigned int line, const char * condMsg, const char * userMsg) Line 103	C++
 	torch_cpu.dll!c10::IListRefIterator<at::Tensor>::~IListRefIterator<at::Tensor>() Line 405	C++
 	torch_cpu.dll!c10::IListRef<at::Tensor>::materialize() Line 571	C++
 	torch_cpu.dll!at::meta::structured_cat::meta(c10::IListRef<at::Tensor> tensors, __int64 dim) Line 63	C++
 	torch_cuda_cu.dll!at::`anonymous namespace'::wrapper_cat(c10::ArrayRef<at::Tensor> tensors, __int64 dim) Line 5172	C++
 	torch_cuda_cu.dll!c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::ArrayRef<at::Tensor>,__int64),&at::`anonymous namespace'::wrapper_cat>,at::Tensor,c10::guts::typelist::typelist<c10::ArrayRef<at::Tensor>,__int64>>::operator()(c10::ArrayRef<at::Tensor> <args_0>, __int64 <args_1>) Line 13	C++
 	torch_cuda_cu.dll!c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::ArrayRef<at::Tensor>,__int64),&at::`anonymous namespace'::wrapper_cat>,at::Tensor,c10::guts::typelist::typelist<c10::ArrayRef<at::Tensor>,__int64>>,at::Tensor __cdecl(c10::ArrayRef<at::Tensor>,__int64)>::call(c10::OperatorKernel * functor, c10::DispatchKeySet __formal, c10::ArrayRef<at::Tensor> <args_0>, __int64 <args_1>) Line 453	C++
 	torch_cpu.dll!c10::callUnboxedKernelFunction<at::Tensor,c10::ArrayRef<at::Tensor>,__int64>(void * unboxed_kernel_func, c10::OperatorKernel * functor, c10::DispatchKeySet dispatchKeySet, c10::ArrayRef<at::Tensor> && <args_0>, __int64 && <args_1>) Line 57	C++
 	torch_cpu.dll!c10::KernelFunction::call<at::Tensor,c10::ArrayRef<at::Tensor>,__int64>(const c10::OperatorHandle & opHandle, c10::DispatchKeySet dispatchKeySet, c10::ArrayRef<at::Tensor> <args_0>, __int64 <args_1>) Line 67	C++
 	torch_cpu.dll!c10::Dispatcher::redispatch<at::Tensor,c10::ArrayRef<at::Tensor>,__int64>(const c10::TypedOperatorHandle<at::Tensor __cdecl(c10::ArrayRef<at::Tensor>,__int64)> & op, c10::DispatchKeySet currentDispatchKeySet, c10::ArrayRef<at::Tensor> <args_0>, __int64 <args_1>) Line 545	C++
 	torch_cpu.dll!c10::TypedOperatorHandle<at::Tensor __cdecl(c10::ArrayRef<at::Tensor>,__int64)>::redispatch(c10::DispatchKeySet currentDispatchKeySet, c10::ArrayRef<at::Tensor> <args_0>, __int64 <args_1>) Line 419	C++
 	torch_cpu.dll!at::_ops::cat::redispatch(c10::DispatchKeySet dispatchKeySet, c10::ArrayRef<at::Tensor> tensors, __int64 dim) Line 1774	C++
 	torch_cpu.dll!at::redispatch::cat(c10::DispatchKeySet dispatchKeySet, c10::ArrayRef<at::Tensor> tensors, __int64 dim) Line 1352	C++
 	torch_cpu.dll!torch::autograd::VariableType::`anonymous-namespace'::cat::__l2::<lambda>() Line 3245	C++
 	torch_cpu.dll!torch::autograd::VariableType::`anonymous namespace'::cat(c10::DispatchKeySet ks, c10::ArrayRef<at::Tensor> tensors, __int64 dim) Line 3243	C++
 	torch_cpu.dll!c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>::operator()(c10::DispatchKeySet <args_0>, c10::ArrayRef<at::Tensor> <args_1>, __int64 <args_2>) Line 13	C++
 	torch_cpu.dll!c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>,at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64)>::call(c10::OperatorKernel * functor, c10::DispatchKeySet dispatchKeySet, c10::ArrayRef<at::Tensor> <args_0>, __int64 <args_1>) Line 470	C++
 	torch_cpu.dll!c10::impl::call_functor_with_args_from_stack_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>,0,0,1,c10::ArrayRef<at::Tensor>,__int64>(c10::OperatorKernel * functor, c10::DispatchKeySet dispatchKeySet, std::vector<c10::IValue,std::allocator<c10::IValue>> * stack, std::integer_sequence<unsigned __int64,0,1> __formal, c10::guts::typelist::typelist<c10::ArrayRef<at::Tensor>,__int64> * __formal) Line 488	C++
 	torch_cpu.dll!c10::impl::call_functor_with_args_from_stack<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>,0>(c10::OperatorKernel * functor, c10::DispatchKeySet dispatchKeySet, std::vector<c10::IValue,std::allocator<c10::IValue>> * stack) Line 503	C++
 	torch_cpu.dll!<lambda_4cbed2dd7523596fd1d34a47d2dbea59>::operator()<c10::guts::detail::_identity>(c10::guts::detail::_identity delay_check) Line 576	C++
 	torch_cpu.dll!c10::guts::detail::_if_constexpr<1>::call<<lambda_4cbed2dd7523596fd1d34a47d2dbea59>,<lambda_4be43825ec1e3472a2677815c3f3c0c2>,0>(c10::impl::make_boxed_from_unboxed_functor<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>,0>::call::__l2::<lambda_4cbed2dd7523596fd1d34a47d2dbea59> && thenCallback, c10::impl::make_boxed_from_unboxed_functor<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>,0>::call::__l2::<lambda_4be43825ec1e3472a2677815c3f3c0c2> && __formal) Line 227	C++
 	torch_cpu.dll!c10::guts::if_constexpr<1,<lambda_4cbed2dd7523596fd1d34a47d2dbea59>,<lambda_4be43825ec1e3472a2677815c3f3c0c2>>(c10::impl::make_boxed_from_unboxed_functor<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>,0>::call::__l2::<lambda_4cbed2dd7523596fd1d34a47d2dbea59> && thenCallback, c10::impl::make_boxed_from_unboxed_functor<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>,0>::call::__l2::<lambda_4be43825ec1e3472a2677815c3f3c0c2> && elseCallback) Line 355	C++
 	torch_cpu.dll!c10::impl::make_boxed_from_unboxed_functor<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor __cdecl(c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64),&torch::autograd::VariableType::`anonymous namespace'::cat>,at::Tensor,c10::guts::typelist::typelist<c10::DispatchKeySet,c10::ArrayRef<at::Tensor>,__int64>>,0>::call(c10::OperatorKernel * functor, const c10::OperatorHandle & __formal, c10::DispatchKeySet dispatchKeySet, std::vector<c10::IValue,std::allocator<c10::IValue>> * stack) Line 592	C++
 	torch_cpu.dll!c10::KernelFunction::callBoxed(const c10::OperatorHandle & opHandle, c10::DispatchKeySet dispatchKeySet, std::vector<c10::IValue,std::allocator<c10::IValue>> * stack) Line 51	C++
 	torch_cpu.dll!c10::Dispatcher::callBoxed(const c10::OperatorHandle & op, std::vector<c10::IValue,std::allocator<c10::IValue>> * stack) Line 572	C++
 	torch_cpu.dll!c10::OperatorHandle::callBoxed(std::vector<c10::IValue,std::allocator<c10::IValue>> * stack) Line 362	C++
 	torch_cpu.dll!c10::OperatorHandle::callBoxed(std::vector<c10::IValue,std::allocator<c10::IValue>> & stack) Line 366	C++
 	torch_cpu.dll!torch::jit::`anonymous-namespace'::createOperatorFromC10::__l2::<lambda>(std::vector<c10::IValue,std::allocator<c10::IValue>> & stack) Line 15	C++
 	[External Code]	
 	torch_cpu.dll!torch::jit::Operation::operator()(std::vector<c10::IValue,std::allocator<c10::IValue>> & stack) Line 42	C++
 	torch_cpu.dll!torch::jit::InterpreterStateImpl::runImpl(std::vector<c10::IValue,std::allocator<c10::IValue>> & stack) Line 294	C++
 	torch_cpu.dll!torch::jit::InterpreterStateImpl::run(std::vector<c10::IValue,std::allocator<c10::IValue>> & stack) Line 1009	C++
 	torch_cpu.dll!torch::jit::InterpreterState::run(std::vector<c10::IValue,std::allocator<c10::IValue>> & stack) Line 1132	C++
 	torch_cpu.dll!torch::jit::GraphExecutorImplBase::run(std::vector<c10::IValue,std::allocator<c10::IValue>> & stack) Line 583	C++
 	torch_cpu.dll!torch::jit::GraphExecutor::run(std::vector<c10::IValue,std::allocator<c10::IValue>> & inputs) Line 827	C++
 	torch_cpu.dll!torch::jit::GraphFunction::run(std::vector<c10::IValue,std::allocator<c10::IValue>> & stack) Line 65	C++
 	torch_cpu.dll!torch::jit::Function::operator()(std::vector<c10::IValue,std::allocator<c10::IValue>> stack, const std::unordered_map<std::string,c10::IValue,std::hash<std::string>,std::equal_to<std::string>,std::allocator<std::pair<std::string const ,c10::IValue>>> & kwargs) Line 63	C++
 	torch_cpu.dll!torch::jit::Method::operator()(std::vector<c10::IValue,std::allocator<c10::IValue>> stack, const std::unordered_map<std::string,c10::IValue,std::hash<std::string>,std::equal_to<std::string>,std::allocator<std::pair<std::string const ,c10::IValue>>> & kwargs) Line 209	C++
 	torch_cpu.dll!torch::jit::Module::forward(std::vector<c10::IValue,std::allocator<c10::IValue>> inputs, const std::unordered_map<std::string,c10::IValue,std::hash<std::string>,std::equal_to<std::string>,std::allocator<std::pair<std::string const ,c10::IValue>>> & kwargs) Line 114	C++

[Issue_1](https://discuss.pytorch.org/t/libtorch-crash-in-win10/139234/8) and [Issue_2](https://github.com/pytorch/pytorch/issues/49460) are somehow related but there was no solution there.

### Versions

System:
- Windows 10
- Visual Studio Enterprise 2019
- Libtorch 1.12.1
- CUDA 11.3

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm

**Comment by yandongwei** [comment]
same problem too！

**Comment by gellston** [comment]
same problem too ㅠㅠ somebody know solution to fix it ? (If someone know short term solution, it's okay)
I also reported same issue, same code and dump file in issue list
cc @peterjc123

**Comment by mszhanyi** [comment]
Could you try debugging Pytorch C++ code https://github.com/pytorch/pytorch/wiki/Best-Practices-to-Edit-and-Compile-Pytorch-Source-Code-On-Windows

**Comment by gellston** [comment]
@mszhanyi  I will try that tomorrow, also i checked that (libtorch 11.0 + cuda11.3) combination works. I guess something has changed between the 11.0 and 11.6 versions.

**Comment by iremyux** [comment]
@rodrigovimieiro @yandongwei @gellston 

Hi,

I am trying to reproduce the error myself in order to have a better understanding of the problem. Even though I am using the same versions and such, I cannot seem to get an error in both Release and Debug modes. 

Can you please provide me more details on how to reproduce issue or provide access to some environment where I can reproduce it?

**Comment by iremyux** [comment]
@rodrigovimieiro 

I am unable to create this issue even though I have created the same environment with the details you've given. This is the reason why I asked for extra details on the issue. 

On the dialog window that you've shared, there is actually a 'Retry' button, which takes you to Debug in VS. I kindly ask you to press 'Retry'', save the dump file under Debug menu of VS and share it with me. 

Also, the UNet.pt file or the source code are not provided in the description. Since this seems to be only different part that we have, it would be quite helpful if you could share it.

**Comment by vladimir-aubrecht** [comment]
@rodrigovimieiro Any chance you could provide as step by step guide on how to reproduce it? @iremyux tried to reproduce it with versions of software you provided, but unfortunately no luck.

**Comment by iremyux** [comment]
I have found out that the issue is fixed at [#78636](https://github.com/pytorch/pytorch/pull/78636). Unfortunately, the library was built with earlier versions. If you build your own libtorch from the latest resources, it should work.
